src/io/flutter/view/FlutterView.java

```diff
@@ -254,27 +256,33 @@ protected void perform(AnActionEvent event) {
 }
 
 class TogglePlatformAction extends AbstractFlutterAction {
-  private boolean isAndroid = false;
-
   TogglePlatformAction(@NotNull FlutterView view) {
     super(view, FlutterBundle.message("flutter.view.togglePlatform.text"), FlutterBundle.message("flutter.view.togglePlatform.description"),
           AllIcons.RunConfigurations.Application);
   }
 
   @Override
   public void actionPerformed(AnActionEvent event) {
-    isAndroid = !isAndroid;
-
     // ext.flutter.platformOverride, value: iOS, android
-    final Map<String, Object> params = new HashMap<>();
-    params.put("value", isAndroid ? "android" : "iOS");
-    view.getFlutterApp().callServiceExtension("ext.flutter.platformOverride", params);
+    final CompletableFuture<JsonObject> result = view.getFlutterApp().callServiceExtension(
```

Comment:
Maybe move the implementation to a method on FlutterApp?

view.getFlutterApp().togglePlatform().thenAccept((boolean isAndroid) -> {
  ... print console message ...
});